diff --git a/dist/index.js b/dist/index.js
index b7eacf17711de6d5d476a8f9d4343809ded1f46b..2eb23eb2503d534e36c40f0578ced4d904960418 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -27,7 +27,7 @@ const tanstackViteConfig = (options) => {
 				exclude: options.exclude,
 				tsconfigPath: options.tsconfigPath,
 				compilerOptions: {
-					module: 199,
+					module: 99,
 					declarationMap: false
 				},
 				beforeWriteFile: (filePath, content) => {
@@ -54,7 +54,7 @@ const tanstackViteConfig = (options) => {
 				exclude: options.exclude,
 				tsconfigPath: options.tsconfigPath,
 				compilerOptions: {
-					module: 199,
+					module: 100,
 					declarationMap: false
 				},
 				beforeWriteFile: (filePath, content) => {
diff --git a/src/index.ts b/src/index.ts
index 74d87604e414f282589fc2df278040a10e7901e7..0b16ec25c9849de8905fb96ea889e3d8d6e325ae 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -57,7 +57,7 @@ export const tanstackViteConfig = (options: Options): UserConfig => {
         exclude: options.exclude,
         tsconfigPath: options.tsconfigPath,
         compilerOptions: {
-          module: 199, // NodeNext
+          module: 99, // ESNext
           declarationMap: false,
         },
         beforeWriteFile: (filePath, content) => {
@@ -83,7 +83,7 @@ export const tanstackViteConfig = (options: Options): UserConfig => {
             exclude: options.exclude,
             tsconfigPath: options.tsconfigPath,
             compilerOptions: {
-              module: 199, // NodeNext
+              module: 100, // Node16
               declarationMap: false,
             },
             beforeWriteFile: (filePath, content) => {
